wolfcrypt/src/integer.c

```diff
@@ -371,7 +360,7 @@ int mp_grow (mp_int * a, int size)
   mp_digit *tmp;
 
   /* if the alloc size is smaller alloc more ram */
-  if (a->alloc < size) {
+  if (a->alloc < size || size == 0) {
```

Comment:
Same as mp_copy reason, except here we are trying to the case where mp_grow(a, 0).